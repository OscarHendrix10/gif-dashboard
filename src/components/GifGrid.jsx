import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category }) => {


    const {images, isLoading} = useFetchGifs( category );

    return (
    <>
        <h3> {category} </h3>
        {
            isLoading && <p>Loading...</p>
        }
        

        <div className="card-grid">
            {
                images.map( (image) => (
                    <GifItem 
                    key={ image.id } 
                    { ...image}
                    />
                ))
            }
        </div>
    
      
    </>
  )
}

