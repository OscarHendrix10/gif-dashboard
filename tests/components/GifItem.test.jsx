import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('pruesbas en gifitema', () => { 

    const title = "saitama";
    const url = "https://localhost/algo.jpg";

    test('debe hacer match con el snapshot', () => {
        
        

        const { container } = render( <GifItem title={ title } url={ url } /> );

        expect(container).toMatchSnapshot();

    });


    test('debe de mostrar la imagen con el URL y el alt Indicado', () => {
            
            
            render( <GifItem title={ title } url={ url } /> );
            // screen.debug();

            // expect(screen.getByRole('img').src).toBe(url);

            const {src, alt} = screen.getByRole('img');
            expect(src).toBe(url);
            expect(alt).toBe(alt);
        });

    test('debe de mostrar el texto', () => {

        render( <GifItem title={ title } url={ url } /> );

        expect(screen.getByText(title)).toBeTruthy();


    });


    });