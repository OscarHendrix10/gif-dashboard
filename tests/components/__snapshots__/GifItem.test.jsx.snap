// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruesbas en gifitema debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://localhost/algo.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
